### ISSUE TITLE
kiali pod keeps restarting every 60 seconds

### ISSUE THREAD

**Issue by samene** [kiali/kiali#3251]
I have installed kiali operator v1.23

This is my CR

``` 
deployment:
    image_version: v1.23.0
    namespace: istio-system
    verbose_mode: "1"
    view_only_mode: false
external_services:
    grafana:
      url: ""
    prometheus:
      url: http://prometheus-service.cattle-prometheus.svc:8080
    tracing:
      in_cluster_url: http://simple-prod-query.jaeger-operator.svc:16686
      url: ""
installation_tag: Klerity Kiali
istio_namespace: istio-system
```

After applying the CR, the kiali pods comes up, runs for around a minute and then keeps restarting. They keep restarting forever after that.
I'm attaching the kiali operator logs

**Comment by samene** [comment]
[kiali-operator.log](https://github.com/kiali/kiali/files/5275960/kiali-operator.log)

**Comment by jmazzitelli** [comment]
Sounds like a duplicate of https://github.com/kiali/kiali/issues/3231. A fix has been applied and will be in the next release. I'll look at the attached operator logs to see if its the same thing.

**Comment by jmazzitelli** [comment]
I only see two reconciliation loops in the log file, so I can't say for sure. I thought the bug in #3231 was only going to happen if you use accessible_namespace of "[**]" but in this case the default value is used. But it is quite possible the same thing happens because of the regex expression - that also processed the list of namespaces and if the list of namespaces comes back unordered, the same problem will occur.

@samene  Can you look at the Kiali ConfigMap between each "loop" and compare the values of the `deployment.accessible_namespaces` setting? See if the order of namespaces changed after each loop.

**Comment by samene** [comment]
The configmap did not change between pod restarts. Attaching kubectl describe output for reference.

[kiali-configmap.txt](https://github.com/kiali/kiali/files/5276400/kiali-configmap.txt)

**Comment by jmazzitelli** [comment]
Oh, wait.. is it just the pod restarting? I'm under the impression the operator is continually executing a reconciliation loop that causes it. But now that I remember I only saw 2 reconciliation loops in the operator logs (which is expected), now I think it is not what I am thinking of.

What does the Events log say in your cluster? Is there an explanation why it keeps restarting your pods? Like liveness or readiness probes failing?

**Comment by jmazzitelli** [comment]
@samene also, check the kiali pod's logs see if you can catch what it is doing just prior to the restart.

**Comment by samene** [comment]
[kiali-operator.txt](https://github.com/kiali/kiali/files/5278128/kiali-operator.txt)
[kiali-operator-logs.txt](https://github.com/kiali/kiali/files/5278129/kiali-operator-logs.txt)
[replicaset.txt](https://github.com/kiali/kiali/files/5278130/replicaset.txt)
[deployment.txt](https://github.com/kiali/kiali/files/5278131/deployment.txt)
[events.txt](https://github.com/kiali/kiali/files/5278132/events.txt)
[kiali-configmap.txt](https://github.com/kiali/kiali/files/5278133/kiali-configmap.txt)
[kiali-logs.txt](https://github.com/kiali/kiali/files/5278134/kiali-logs.txt)

Attaching all files as requested.

**Comment by jmazzitelli** [comment]
Looking at that new kiali-operator-logs.txt just posted, I'm fairly confident now that this is the same thing as https://github.com/kiali/kiali/issues/3231. The accessible_namespaces list is not ordered and each reconciliation run it is sorted in a different order. So, this will be fixed in the next release since that bug has been fixed and merged into master.

In the meantime, the workaround is to explicitly set `deployment.accessible_namespaces` to the list of namespaces you want Kiali to be given access to. Based on your logs, I think you will want this in your Kiali CR:

```
...
deployment:
  accessible_namespaces: ["istio-system", "metallb-system", "security-scan", "jaeger-operator", "default", "sample", "cattle-system", "cattle-prometheus"]
...
```

Or just set that to a list of namespaces that you want. The namespaces should be those that are in your Istio mesh.

**Comment by jmazzitelli** [comment]
@samene I am curious - what k8s cluster (vendor and version) are you deploying this in? I have not seen this behavior, but you are now the second person who has seen this, and I would like to know what cluster environment you are using (minikube? GKE? other?). Just wondering if it's specific to a particular k8s implementation or if it's a common issue across platforms.

**Comment by samene** [comment]
Actually the problem was that there was another kiali CR deployed in default namespace.  I now set the watchNamespace in the helm chart to only watch for  the kiali-operator namespace.

**Comment by jmazzitelli** [comment]
Oh! Yeah, I never thought about that - multiple Kiali CRs attempting to install Kiali in the same deployment namespace.

**Comment by samene** [comment]
Just a note here, I was facing the same issue again, even with only a single Kiali CR. The kiali pods kept restarting. I can confirm it is because of #3231 since when I checked the configmap between restarts the accessible namespaces kept reordering randomly.

before restart
      accessible_namespaces:
      - istio-system
      - cattle-prometheus
      - security-scan
      - metallb-system
      - cattle-system
      - jaeger-operator
      - sample
      - cert-manager
      - default
      - enterka-cloud

after restart
      accessible_namespaces:
      - istio-system
      - cattle-system
      - metallb-system
      - sample
      - cert-manager
      - jaeger-operator
      - enterka-cloud
      - default
      - security-scan
      - cattle-prometheus

Which release is #3231 fix available?

**Comment by jshaughn** [comment]
@samene The fix is available with the v1.25 kiali operator, to be released tomorrow, Oct 9th.
